### PR TITLE
Add support for FreeIPA replica servers

### PR DESCRIPTION
**Pull request by jsf9k**
Once this PR is merged I will split the master and replica creation into two separate roles.  That will duplicate a little bit of code, but each of the resulting modules will be much simpler than the current one.  This is because there are lots of input variables that are currently only valid and/or required for one case or the other.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,33 +69,31 @@ module "ipa" {
...
+| domain | The domain for the IPA server (e.g. `example.com`).  Only required if this is a master IPA server (i.e. if is_master is true). | string | Empty string | no |
+| hostname | The hostname of this IPA server (e.g. `ipa.example.com`) | string | | yes |
+| is_master | Indicates whether this IPA server is a master (true) or a replica (false) | bool | | yes |
+| master_hostname | The hostname of the IPA master (e.g. ipa.example.com).  Only necessary if creating a replica IPA server and you want the replica to delay installation until the master is available. | string | Empty string | no |
+| private_reverse_zone_id | The zone ID corresponding to the private Route53 reverse zone where the PTR records related to this IPA server should be created (e.g. `ZKX36JXQ8W82L`) | string | | yes |
+| private_zone_id | The zone ID corresponding to the private Route53 zone where the kerberos-related DNS records should be created (e.g. `ZKX36JXQ8W82L`) | string | | yes |
+| public_zone_id | The zone ID corresponding to the public Route53 zone where the kerberos-related DNS records should be created (e.g. `ZKX36JXQ8W82L`) | string | Empty string | no |
```
Comment (dav3r) [inline README.md]: "Kerberos" should be capitalized in both the `private_zone_id` and `public_zone_id` descriptions

File: dns/route53_zone/variables.tf [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  description = "The IP of this IPA server (e.g. 10.11.1.5)"
+}
+
+variable "is_master" {
+  type        = bool
+  description = "Indicates whether this IPA server is a master (true) or a replica (false)"
+}
+
+variable "zone_id" {
+  description = "The zone ID corresponding to the Route53 zone where the kerberos-related DNS records should be created (e.g. ZKX36JXQ8W82L)"
```
Comment (dav3r) [inline dns/route53_zone/variables.tf]: Capitalize Kerberos here also

File: security_group.tf [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  security_group_id = aws_security_group.ipa_servers[0].id
+  type              = "ingress"
+  protocol          = "tcp"
+  self              = true
+  from_port         = local.ipa_tcp_ports[count.index]
+  to_port           = local.ipa_tcp_ports[count.index]
+}
+
+# TCP egress rules for IPA
+resource "aws_security_group_rule" "ipa_tcp_egress" {
```
Comment (dav3r) [inline security_group.tf]: Should this rule be renamed `ipa_tcp_egress_self` for consistency with `ipa_tcp_ingress_self`?  Same thing below for `ipa_udp_egress_self`.

File: variables.tf [hunk 4]
```diff
@@ -5,61 +5,81 @@
...
 }
 
-variable "public_zone_id" {
-  description = "The zone ID corresponding to the public Route53 zone where the kerberos-related DNS records should be created (e.g. ZKX36JXQ8W82L)"
-}
-
-variable "realm" {
-  description = "The realm for the IPA server (e.g. EXAMPLE.COM)"
+variable "private_zone_id" {
+  description = "The zone ID corresponding to the private Route53 zone where the kerberos-related DNS records should be created (e.g. ZKX36JXQ8W82L)"
```
Comment (dav3r) [inline variables.tf]: Capitalize Kerberos here again

**Review by dav3r — APPROVED** [review]
This looks great to me - awesome work @jsf9k!  I only have very superficial comments on what you have done here.

**Review by mcdonnnj — APPROVED** [review]
This is some really nice work @jsf9k , I don't see anything of note beyond what @dav3r has already said.

File: README.md (hunk 1 above)
Comment (jsf9k) [reply README.md]: Fixed in 87d754c.

File: variables.tf (hunk 4 above)
Comment (jsf9k) [reply variables.tf]: Fixed in 87d754c.

File: dns/route53_zone/variables.tf (hunk 2 above)
Comment (jsf9k) [reply dns/route53_zone/variables.tf]: Fixed in 87d754c.

File: security_group.tf (hunk 3 above)
Comment (jsf9k) [reply security_group.tf]: Corrected in 8db8976.

**Pull request merged**